### PR TITLE
Reorganize Getting Started into Zero-to-Robot

### PR DESCRIPTION
**Pull request by Daltz333**
This PR restructures Getting Started into more bite-sized Zero-to-Robot. This allows us better control about the content and expansiveness of the articles.

This PR supersedes #796.

Rewriting the articles will come other PRs that depend on this. I thought it would be best to have the restructuring standalone.

Preview Link: https://frc-docs--861.org.readthedocs.build/en/861/

Changes:
- Renamed Getting-Started into Zero-to-Robot
- Converted "Intro.rst" into "Introduction.rst" before all other content
- Split up content into 3 steps, Building, Dependencies, Programming
- Merged CAN article into Step 2 index (no need for a separate article)
- Fixed any bad references throughout the docs that point to the old URLs

### CODE REVIEW

File: source/docs/zero-to-robot/step-2/index.rst [hunk 1]
```diff
@@ -1,4 +1,15 @@
-Updating and Configuring Pneumatics Control Module and Power Distribution Panel
-===============================================================================
+Step 2: Installing Dependencies
```
Comment (sciencewhiz) [inline source/docs/zero-to-robot/step-2/index.rst:1]: I don't like the term Dependencies here. Dependencies for what? How about just Installing Software?

**Review by sciencewhiz — CHANGES_REQUESTED** [review]

File: source/docs/zero-to-robot/step-2/index.rst [hunk 2]
```diff
@@ -1,4 +1,15 @@
...
+===============================
 
 .. warning:: The CAN Web Dashboard plugin is no longer supported by the roboRIO Web Dashboard. To configure CTRE CAN devices such as the PCM and PDP, use `CTRE Phoenix Tuner <https://phoenix-documentation.readthedocs.io/en/latest/ch03_PrimerPhoenixSoft.html#what-is-phoenix-tuner>`_
```
Comment (Starlight220) [inline source/docs/zero-to-robot/step-2/index.rst:4]: Why is this warning here, and not in the `configuring CAN devices` section/page?

Comment (Daltz333) [reply source/docs/zero-to-robot/step-2/index.rst:4]: The page was a stub and isn't really needed. I'm considering adding a 3rd party page that has the mention.

File: source/docs/zero-to-robot/step-2/index.rst (hunk 1 above)
Comment (Daltz333) [reply source/docs/zero-to-robot/step-2/index.rst:1]: Sure

File: source/docs/zero-to-robot/introduction.rst [hunk 3]
```diff
@@ -19,18 +25,12 @@ C++
...
-
-- `Code Academy <https://www.codecademy.com/learn/learn-java>`__
-- `Head First Java 2nd Edition <https://www.amazon.com/dp/0596009208/>`__ is a very beginner friendly introduction to programming in Java (ISBN-10: 0596009208).
-
 LabVIEW
 ^^^^^^^
 
 - `NI Learn LabVIEW <http://www.ni.com/academic/students/learn-labview/>`__
 
 0 to Robot
```
[the remark this answers is not in the document]
Comment (Daltz333) [reply source/docs/zero-to-robot/introduction.rst:33]: Will do

**Comment by Starlight220** [comment]
While reviewing this I'm seeing quite a few problems that are in `latest`. I submit PRs to correct them in `latest`, but will those changes take affect here?
(I'm talking about #864 , #863 and possibly others)

**Comment by Daltz333** [comment]
Yes. I will rebase

**Comment by Starlight220** [comment]
I think a simple `git rebase` might not apply the changes as this PR renames some files. I suggest to simply make sure all of the problems fixed by my small PRs are fixed before merging this - I'll update my above comment with any PRs I add.

**Comment by Daltz333** [comment]
A rebase does apply the changes assuming I make no other changes

**Comment by Starlight220** [comment]
I knew that you meant you would rebase this PR on those, I just wasn't sure that it would track the changes through the file rename

**Comment by Starlight220** [comment]
The linkage to the `Deploying Robot Code` page [here](https://frc-docs--861.org.readthedocs.build/en/861/docs/zero-to-robot/step-3/creating-benchtop-test-program-cpp-java.html#deploying-the-project-to-a-robot) breaks the flow a bit. I suggest we provide a shortened version of the build/deploy article in the 0-robot flow (independent of the full article, but linking to it) or put the full build/deploy page (that's actually not very long) in the 0-robot flow. As the article is relatively short I think the latter is better.

**Comment by Starlight220** [comment]
Another small suggestion: maybe add a short section that finalizes the 0-robot track? I found the transition from part 3 of 0-Robot to `what is wpilib` a bit abrupt.

**Comment by Daltz333** [comment]
This PR is strictly for renaming and reordering. Additionally changes should be in other PRs

**Review by Kevin-OConnor — APPROVED** [review]
LGTM

**Review by AustinShalit — APPROVED** [review]

**Pull request merged**